From 89036f62b573bc2df30e7b865da65454213f6720 Mon Sep 17 00:00:00 2001
From: Matthias Reichl <hias@horus.com>
Date: Sun, 11 Oct 2020 00:48:55 +0200
Subject: [PATCH] bcm2835-dma: only reserve channel 0 if legacy dma
 driver is enabled

If CONFIG_DMA_BCM2708 isn't enabled there's no need to mask out
one of the already scarce DMA channels.

Signed-off-by: Matthias Reichl <hias@horus.com>
---
 drivers/dma/bcm2835-dma.c | 2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/dma/bcm2835-dma.c
+++ b/drivers/dma/bcm2835-dma.c
@@ -1283,6 +1283,7 @@ static int bcm2835_dma_probe(struct plat
 		goto err_no_dma;
 	}
 
+#ifdef CONFIG_DMA_BCM2708
 	/* One channel is reserved for the legacy API */
 	if (chans_available & BCM2835_DMA_BULK_MASK) {
 		rc = bcm_dmaman_probe(pdev, base,
@@ -1293,6 +1294,7 @@ static int bcm2835_dma_probe(struct plat
 
 		chans_available &= ~BCM2835_DMA_BULK_MASK;
 	}
+#endif
 
 	/* And possibly one for the 40-bit DMA memcpy API */
 	if (chans_available & od->cfg_data->chan_40bit_mask &
